Makefile

```diff
@@ -737,6 +737,9 @@ docker-build-all: $(addprefix docker-build-,$(ALL_ARCH)) ## Build docker images
 docker-build-%:
 	$(MAKE) ARCH=$* docker-build
 
+docker-push-%: docker-build-%
+	$(MAKE) ARCH=$* docker-push
```

Comment:
I'll note that a disadvantage of this is that doing an explicit push will *always* do a build first. There may be manual workflows where this is annoying, but hopefully folks are pushing from a manual workflow and this is most likely safer even if they are.

Reply:
Appreciate the call out -- I'll leave this comment thread open while waiting on review from maintainers

Reply:
Can we assess what the speed improvement of this change is? I wonder if it's worth making the push target confusing like this. 

If there's a significant improvement it might be better to make this a new target so folks still have targets to simply push images to the registry.

Reply:
If I read it right there are three variants:

1. old state (sequential build + push separate, no parallelism)
2. immediate state (sequential build + push, including parallelism on sequential steps (build + push))
3. current state of this PR (build + push + parallelism all in parallel)

Doing 3 means: there could be a build which fails and then we partially push images which succeeded building right?

How much is the time gain between 2 and 3?

Reply:
yes that looks like an accurate summary @chrischdi 

> Doing 3 means: there could be a build which fails and then we partially push images which succeeded building right?

Yes that does seem like a risk that would be introduced by this change 

> Can we assess what the speed improvement of this change is? I wonder if it's worth making the push target confusing like this.
> 
> If there's a significant improvement it might be better to make this a new target so folks still have targets to simply push images to the registry.

on this most recent run i ran with variant 2, there was no difference in run time from the last test run with variant 3 (in fact it completed in slightly less time):
https://a.cl.ly/RBuzw2j7 

Given this finding and the risk brought up with a failing build and partial push, I'll go ahead switch back to 3 sub-makes within `release-staging` and revert to the original `docker-push-%` target.  

Sound reasonable @killianmuldoon ?

Reply:
Sounds good! Thanks :slightly_smiling_face: